lib/types/command_payload_json.ex

```diff
@@ -0,0 +1,206 @@
+    ExecPayload,
+    MetaData,
+    NetworkID,
+    OptionalCapsList,
+    PactCode,
+    PactDecimal,
+    PactInt,
+    PactPayload,
+    PactTransactionHash,
+    PactValue,
+    PactValuesList,
+    Proof,
+    Rollback,
+    Signer,
+    SignersList,
+    Step
+  }
+
+  @behaviour Kadena.Types.Spec
+
+  @type json :: String.t()
+  @type signers_list :: SignersList.t()
+  @type signers :: list(map())
+  @type network_id :: NetworkID.t()
+  @type str :: String.t()
+  @type pact_payload :: PactPayload.t()
+  @type proof :: Proof.t() | nil
+  @type data :: EnvData.t() | nil
+  @type meta :: MetaData.t()
+  @type signer :: Signer.t()
+  @type addr :: Base16String.t() | nil
+  @type scheme :: :ed25519 | nil
+  @type valid_scheme :: :ED25519 | nil
+  @type clist :: CapsList.t() | nil
+  @type valid_clist :: list(map())
+  @type cap :: Cap.t()
+  @type pact_values_list :: PactValuesList.t()
+  @type literal ::
+          integer()
+          | boolean()
+          | String.t()
+          | PactInt.t()
+          | PactDecimal.t()
+          | PactValuesList.t()
+  @type value :: integer() | str() | boolean() | Decimal.t()
+  @type valid_pact_value :: list(value)
+  @type pact_value :: list(valid_pact_value())
+
+  @type t :: %__MODULE__{json: json()}
+
+  defstruct [:json]
+
+  @impl true
+  def new(%CommandPayload{
+        network_id: network_id,
+        payload: payload,
+        signers: signers,
+        meta: meta,
+        nonce: nonce
+      }) do
+    with {:ok, payload} <- extract_payload(payload),
+         {:ok, meta} <- extract_meta(meta),
+         {:ok, network_id} <- extract_network_id(network_id),
+         {:ok, signers} <- extract_signers_list(signers) do
+      request_body =
+        %{
+          payload: payload,
+          meta: meta,
+          network_id: network_id,
+          nonce: nonce,
+          signers: signers
+        }
+        |> Case.to_camel!()
+        |> Jason.encode!()
+
+      %__MODULE__{json: request_body}
+    end
+  end
+
+  @spec extract_network_id(network_id()) :: {:ok, str()}
+  defp extract_network_id(%NetworkID{id: id}), do: {:ok, id}
+
+  @spec extract_payload(pact_payload()) :: {:ok, map()}
+  defp extract_payload(%PactPayload{payload: %ExecPayload{} = exec_payload}) do
+    %ExecPayload{code: %PactCode{code: code}, data: data} = exec_payload
+    payload = %{exec: %{code: code, data: extract_data(data)}}
+    {:ok, payload}
+  end
+
+  defp extract_payload(%PactPayload{
+         payload: %ContPayload{
+           data: data,
+           pact_id: %PactTransactionHash{hash: hash},
+           proof: proof,
+           rollback: %Rollback{value: rollback},
+           step: %Step{number: number}
+         }
+       }) do
+    payload = %{
+      cont: %{
+        data: extract_data(data),
+        pact_id: hash,
+        proof: extract_proof(proof),
+        roollback: rollback,
+        step: number
+      }
+    }
+
+    {:ok, payload}
+  end
+
+  @spec extract_proof(proof()) :: str() | nil
```

Comment:
In general, in the project, we are using `types` to define the returns, please use the same pattern. Change `str() | nil`.